openssl-dynamic/src/main/c/ssl.c

```diff
@@ -1897,15 +1897,17 @@ TCN_IMPLEMENT_CALL(jbyteArray, SSL, getSessionId)(TCN_STDARGS, jlong ssl)
 
 TCN_IMPLEMENT_CALL(jint, SSL, getHandshakeCount)(TCN_STDARGS, jlong ssl)
 {
-    int *handshakeCount = NULL;
+    tcn_ssl_state_t *state = NULL;
     SSL *ssl_ = J2P(ssl, SSL *);
 
     TCN_CHECK_NULL(ssl_, ssl, 0);
 
     UNREFERENCED(o);
 
-    handshakeCount = tcn_SSL_get_app_data3(ssl_);
-    return handshakeCount != NULL ? *handshakeCount : 0;
+    if ((state = tcn_SSL_get_app_state(ssl_)) == NULL) {
+        return state->handshakeCount;
```

Comment:
wont state be NULL here by the condition?

Reply:
doh... good catch. This should be `!= NULL`